framework/decode/file_processor.cpp

```diff
@@ -797,6 +797,7 @@ bool FileProcessor::ProcessMetaData(const format::BlockHeader& block_header, for
             {
                 if (decoder->SupportsMetaDataId(meta_data_id))
                 {
+                    decoder->SetCurrentBlockIndex(block_index_);
```

Comment:
Please delete the remaining calls to `SetCurrentBlockIndex`. `FileProcessor::ProcessBlocks()` takes care of calling that function (line 254)